client/components/shared/NewBankForm.js

```diff
@@ -127,72 +127,80 @@ export default class NewBankForm extends React.Component {
             );
 
             let maybeCustomFields = [];
-            if (this.state.hasCustomFields) {
+            if (this.state.customFields.length > 0) {
                 maybeCustomFields = this.state.customFields.map((field, index) =>
-                    <CustomBankField ref={ `customField${index}` } params={ field } />
+                    <CustomBankField
```

Comment:
Wait, it seems that CustomBankField needs `onChange` according to your change in CustomBankField.js, but it isn't there here.